fus.c

```diff
@@ -982,16 +987,49 @@ add_solvables_to_pile (Pool    *pool,
       /* solvables prefixed by @ are file names */
       if (**solvable == '@')
         {
-          if (!add_solvables_from_file_to_pile (*solvable + 1, pool, pile, error))
+          if (!add_solvables_from_file_to_pile (*solvable + 1, pool, pile, exclude, error))
             return FALSE;
         }
       else
-        add_solvable_to_pile (*solvable, pool, pile);
+        add_solvable_to_pile (*solvable, pool, pile, exclude);
     }
 
   return TRUE;
 }
 
+static Queue
+mask_non_default_module_pkgs (Pool *pool)
+{
+  Queue selection;
+  queue_init(&selection);
+
+  Id ndef_modules_rel = pool_rel2id (pool,
+                                     pool_str2id (pool, "module()", 1),
+                                     pool_str2id (pool, "module-default()", 1),
+                                     REL_WITHOUT,
+                                     1);
+  Id *pp = pool_whatprovides_ptr (pool, ndef_modules_rel);
+  for (; *pp; pp++)
+    {
+      Solvable *s = pool_id2solvable (pool, *pp);
+      g_auto(Queue) q;
+      queue_init (&q);
+      Id dep = pool_rel2id (pool, s->name, s->arch, REL_ARCH, 1);
+      pool_whatcontainsdep (pool, SOLVABLE_REQUIRES, dep, &q, 0);
+
+      for (int i = 0; i < q.count; i++)
+        {
+          static int sel_flags = SELECTION_CANON;
```

Comment:
If I had added `SELECTION_ADD` to the flags, I could have removed the Queue and `selection_add` below, simplifying it to 1 line. Not sure how to update this now though.

Reply:
I'm going to fix this in a separate commit.